tests/appsec/iast/_test_iast_fixtures.py

```diff
@@ -101,14 +101,16 @@ def test_telemetry_metric_instrumented_sink(self):
         assert series, f"Got no series for metric {expected_metric}"
         logging.debug("Series: %s", series)
         expected_tag = f"vulnerability_type:{self.vulnerability_type}"
-        series = [s for s in series if expected_tag in s["tags"]]
+        series = [s for s in series if expected_tag in s["tags"]] or [
```

Comment:
it means `series = [serie_a or serie_b]`. So if `series` will be `[serie_a]` if `serie_a` is not empty, else it will be `[serie_b]`.

It's valid python, but a little bit unusual. And looking the code below, I presume it's not what you intend to do.

Reply:
No, the fix adds a case insensitive validation of the tags. Java sends tags like "vulnerability:WEAK_RANDOMNESS" and python tracer transforms tags to lowercase "vulnerability:weak_randomness"

Reply:
```python
# lower the vulnerability_type, as all assertion will be case-insensitive
expected_tag = f"vulnerability_type:{self.vulnerability_type}".lower()

# Filter by taking only series where expected tag is in the list serie.tags (case insentive check)
series = [serie for serie in series if expected_tag in map(str.lower, serie["tags"])]

assert len(series) != 0, f"Got no series for metric {expected_metric} with tag {expected_tag}"

for serie in series:
            assert serie["_computed_namespace"] == expected_namespace
            assert serie["metric"] == expected_metric
            assert serie["common"] is True
            assert serie["type"] == "count"
            assert len(s["points"]) == 1
            p = s["points"][0]
            assert p[1] >= 1
```

I removed the `assert set([tag.lower() for tag in s["tags"]]) == {expected_tag.lower()}` because : 

* it was bugged, it fails on `[expected_tag, any_other_value]`
* it's redundant, as in this for loop, `serie.tags` contains at least one `expected_tag`, thanks to the filter above